ServerCore/Helpers/TeamHelper.cs

```diff
@@ -210,20 +210,48 @@ public static async Task<Tuple<bool, string>> AddMemberAsync(PuzzleServerContext
                                   select app;
             context.TeamApplications.RemoveRange(allApplications);
 
+            // If event has PlayerClasses: Assign the user a random available PlayerClass
+            // This ensures that they have some class when the event starts
+            // Better flow for picking their own class TBD
+            PlayerClass playerClass = null;
+
+            if (Event.HasPlayerClasses)
+            {
+                playerClass = await PlayerClassHelper.GetRandomPlayerClassFromAvailable(context, Event.ID, EventRole, teamId);
+                Member.Class = playerClass;
+            }
+
             context.TeamMembers.Add(Member);
             await TeamHelper.OnTeamMemberChange(context, team);
             await context.SaveChangesAsync();
 
+            string emailBody = $"Have a great time!";
+
+            if (Event.HasPlayerClasses)
+            {
+                if (playerClass != null)
+                {
+                    emailBody = string.Concat($"This event has assigned a {Event.PlayerClassName} for each player. " +
+                        $"You have been assigned the {Event.PlayerClassName} of {playerClass.Name}. " +
+                        $"To change this {Event.PlayerClassName} go to Teams -> My Team on the website. \r\n", emailBody);
+                }
+                else
+                {
+                    emailBody = string.Concat($"This event has assigned a {Event.PlayerClassName} for each player. " +

```

Comment:
IDK how or why you get here

Reply:
Ideally we wouldn't, this is just in case something goes wrong during the process. It's basically a hedge against the huge number of paths we have into registration and joining teams.